airflow/utils/file.py

```diff
@@ -344,6 +344,10 @@ def might_contain_dag(file_path: str, safe_mode: bool, zip_file: zipfile.ZipFile
     :param zip_file: if passed, checks the archive. Otherwise, check local filesystem.
     :return: True, if file might contain DAGs.
     """
+    might_contain_dag_callable = conf.getimport("core", "might_contain_dag_callable")
+    if might_contain_dag_callable:
+        return might_contain_dag_callable(file_path=file_path, safe_mode=safe_mode, zip_file=zip_file)
+
```

Comment:
I like the concept.  One suggestion:  I would move the existing heuristic into a callable and set that as the default value in the config and clean this method up.

Reply:
@ferruzzi great idea. let me do a refactor